plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1356 @@
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+        state = self._module.params['state']
+
+        if state == 'overridden' or state == 'replaced':
+            commands, requests = self._state_replaced_or_overridden(want, have, state)
+        elif state == 'deleted':
+            commands, requests = self._state_deleted(want, have)
+        elif state == 'merged':
+            commands, requests = self._state_merged(want, have)
+
+        return commands, requests
+
+    def new_test_keys(self, want):
+        """ Returns the test keys based on if want contains name value for agentaddress
+        and/or host options
+        :rtype: A list
+        :returns: new test_keys
+        """
+        test_keys = [
+            {'community': {'name': ''}},
+            {'group': {'name': '', 'access': {'security_model': ''}}},
+            {'engine': ''},
+            {'access': {'security_model': ''}},
+            {'user': {'name': ''}},
+            {'view': {'name': ''}}]
+
+        if 'agentaddress' in want:
+            no_name = False
+            for command in want['agentaddress']:
+                if 'name' not in command:
+                    no_name = True
+            if no_name:
+                test_keys.extend([{'agentaddress': {'ip': ''}},
+                                  {'agentaddress': {'vrf': ''}}])
+            else:
+                test_keys.extend([{'agentaddress': {'name': ''}}])
+        if 'host' in want:
+            no_name = False
+            for command in want['host']:
+                if 'name' not in command:
+                    no_name = True
+            if no_name:
+                test_keys.extend([{'host': {'ip': ''}},
+                                  {'host': {'vrf': ''}}])
+            else:
+                test_keys.extend([{'host': {'name': ''}}])
+
+        return test_keys
+
+    def _state_replaced_or_overridden(self, want, have, state):
+        """
+        The command generator when state is overridden
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+
+        if not want:
+            return commands, requests
+
+        diff_want = get_diff(want, have, TEST_KEYS)
+        del_commands = get_diff(have, want, TEST_KEYS)
+        if not diff_want and not del_commands:
+            return commands, requests
+        merged_commands = None
+        merged_request = None
+
+        if del_commands:
```

Comment:
For correct "replaced" state handling, only "replace" configuration sections that are mentioned in "want".

Here is an example of how to do this:

```suggestion
        if state == 'replaced':
            pop_list = []
            for option in del_commands:
                if not want.get(option)
                pop_list.append(option)
            for option in pop_list:
                del_commmands.pop(option)
            if del_commands:
```